Modules/xSharePoint/DSCResources/MSFT_xSPWebAppPolicy/MSFT_xSPWebAppPolicy.psm1

```diff
@@ -56,63 +81,260 @@ function Set-TargetResource
+                                        }
+                                        "Full Read" {
+                                            $policy.PolicyRoleBindings.Add($fullRead)
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    Missing
+                        {
+                            ## Account is missing, add this account
+                            Write-Verbose -Verbose "Adding $user"
+                            $usersettings = GetUserFromCollection $allMembers $user
+                            $newPolicy = $wa.Policies.Add($user, $user)
+                            foreach ($permissionLevel in $usersettings.PermissionLevel) {
+                                switch ($permissionLevel) {
+                                    "Deny All" {
+                                        $newPolicy.PolicyRoleBindings.Add($denyAll)
+                                    }
+                                    "Deny Write" {
+                                        $newPolicy.PolicyRoleBindings.Add($denyWrite)
+                                    }
+                                    "Full Control" {
+                                        $newPolicy.PolicyRoleBindings.Add($fullControl)
+                                    }
+                                    "Full Read" {
+                                        $newPolicy.PolicyRoleBindings.Add($fullRead)
+                                    }
+                                }
+                            }
+                            if ($usersettings.ActAsSystemAccount) {
+                                $newPolicy.IsSystemUser = $usersettings.ActAsSystemAccount
+                            }
+                        }
                 }
+                $wa.Update()
             }
         }
 
-        if ($null -ne $policyObject) {
-            if ($params.ContainsKey("ActAsSystemUser") -eq $true) {
-                $policyObject.IsSystemUser = $params.ActAsSystemUser
+        if ($params.MembersToInclude) {
+            Write-Verbose -Verbose "Processing MembersToInclude parameter"
+
+            $allMembers = @()
+            foreach ($member in $MembersToInclude) {
+                $allMembers += $member
+            }
+
+            if ($params.SetCacheAccountsPolicy) {
+                Write-Verbose -Verbose "Adding Cache Accounts to MembersToInclude parameter"
+                $psuAccount = @{
+                    UserName = $wa.Properties["portalsuperuseraccount"]
+                    PermissionLevel = "Full Read"
+                }
+                $allMembers += $psuAccount
+                
+                $psrAccount = @{
+                    UserName = $wa.Properties["portalsuperreaderaccount"]
+                    PermissionLevel = "Full Read"
+                }
+                $allMembers += $psrAccount
```

Comment:
Same remark here about the cache accounts need claims and default modes added. Can we consolidate this so we don't have this if statement in the script twice though? Not fussed about leaving it here, just noticed the repetition and wanted to call it out.